crates/build/re_types_builder/src/codegen/docs/snippets_ref.rs

```diff
@@ -0,0 +1,525 @@
+            name,
+            name_qualified,
+            path,
+
+            contents,
+            description,
+
+            python,
+            rust,
+            cpp,
+
+            archetypes,
+            components,
+            archetypes_blueprint,
+            components_blueprint,
+            views,
+        };
+
+        // Fill the reverse indices.
+        for (objs, index) in [
+            (&snippet.archetypes, &mut snippets.per_archetype),
+            (&snippet.components, &mut snippets.per_component),
+            (&snippet.views, &mut snippets.per_view),
+            (
+                &snippet.archetypes_blueprint,
+                &mut snippets.per_archetype_blueprint,
+            ),
+            (
+                &snippet.components_blueprint,
+                &mut snippets.per_component_blueprint,
+            ),
+        ] {
+            for obj in objs {
+                index.entry(obj).or_default().push(snippet.clone());
+            }
+        }
+    }
+
+    Ok(snippets)
+}
+
+/// Neatly organized [`Object`]s (archetypes, components, etc).
+#[derive(Debug)]
+struct KnownObjects<'o> {
+    archetypes: BTreeSet<&'o Object>,
+    components: BTreeSet<&'o Object>,
+
+    archetypes_blueprint: BTreeSet<&'o Object>,
+    components_blueprint: BTreeSet<&'o Object>,
+
+    views: BTreeSet<&'o Object>,
+}
+
+impl<'o> KnownObjects<'o> {
+    fn init(objects: &'o Objects) -> Self {
+        let (
+            mut archetypes,
+            mut components,
+            mut archetypes_blueprint,
+            mut components_blueprint,
+            mut views,
+        ) = (
+            BTreeSet::new(),
+            BTreeSet::new(),
+            BTreeSet::new(),
+            BTreeSet::new(),
+            BTreeSet::new(),
+        );
+
+        for object in objects.values() {
+            // skip test-only archetypes
+            if object.is_testing() {
+                continue;
+            }
+
+            match object.kind {
+                ObjectKind::Archetype if object.scope().as_deref() == Some("blueprint") => {
+                    archetypes_blueprint.insert(object);
+                }
+
+                ObjectKind::Archetype => {
+                    archetypes.insert(object);
+                }
+
+                ObjectKind::Component if object.scope().as_deref() == Some("blueprint") => {
+                    components_blueprint.insert(object);
+                }
+
+                ObjectKind::Component => {
+                    components.insert(object);
+                }
+
+                ObjectKind::View => {
+                    views.insert(object);
+                }
+
+                ObjectKind::Datatype => {}
+            };
+        }
+
+        Self {
+            archetypes,
+            components,
+            archetypes_blueprint,
+            components_blueprint,
+            views,
+        }
+    }
+}
+
+// ---
+
+/// Returns `true` if the given name has not been released yet.
```

Comment:
While I find this approach _Good Enough_ :tm:, I'm all ears if someone has something nicer.